urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/httpd/response/successful/OK.java

```diff
@@ -0,0 +1,698 @@
+		
+		// This might throw a MalformedRequestException
+		// (NullPointerException cannot be raised because the header field is not null)
+		ContentRange cRange = ContentRange.parse( request_contentRange );
+		// The numerical fields are in range so far ...
+		// ... but we have to check if the instanceLength is in range (the parser does not
+		//     know the resource's length).
+		if( (cRange.getInstanceLength() != -1 
+		     && cRange.getInstanceLength() != resource.getLength()
+		     )
+		    || cRange.getLastBytePosition() >= resource.getLength() ) {
+		    
+		    // TODO:
+		    //  Server should send a '416 Requested range not satisfiable' error response!
+		    optionalReturnSettings.put( Constants.KEY_GENERATED_STATUS_CODE, 
+						new BasicNumberType(Constants.HTTP_STATUS_CLIENTERROR_REQUEST_RANGE_NOT_SATISFIABLE) );
+		    optionalReturnSettings.put( Constants.KEY_GENERATED_REASON_PHRASE,
+						new BasicStringType("Requested range not satisfiable") );
+		    //throw new MalformedRequestException( "Cannot satisfy requested range (out of bounds)" );
+		    throw new ParametrizedHTTPException( "Cannot satisfy requested range (out of bounds).",
+							 Constants.HTTP_STATUS_CLIENTERROR_REQUEST_RANGE_NOT_SATISFIABLE,
+							 "Requested range not satisfiable" );
+							 
+
+		}
+		
+		// Build a 'ranged' resource.
+		// ... send a '206 Partial Content' response.
+		optionalReturnSettings.put( Constants.KEY_GENERATED_STATUS_CODE, 
+					    new BasicNumberType(Constants.HTTP_STATUS_SUCCESSFUL_PARTIAL_CONTENT) );
+		optionalReturnSettings.put( Constants.KEY_GENERATED_REASON_PHRASE,
+					    new BasicStringType("Partial Content") );
+		long oldResourceLength = resource.getLength();
+		resource = new RangedResource( resource,
+					       cRange,
+					       this.getHTTPHandler(),
+					       this.getHTTPHandler().getLogger()
+					       );
+
+
+		// Replace the OK status!
+		super.setStatusCode( Integer.toString(Constants.HTTP_STATUS_SUCCESSFUL_PARTIAL_CONTENT) );
+		super.setReasonPhrase( "Partial Content" );
+
+		// Following header fields MUST be included in the response:
+		//  - Content-Range
+		//  - Date
+		//  - ETag and/or Content-Location
+		//  - Expires, Cache-Control and/or Vary
+		ContentRange responseRange = new ContentRange( ContentRange.NAME_BYTESUNIT_BYTES,
+							       cRange.getFirstBytePosition(),
+							       cRange.getLastBytePosition(),
+							       oldResourceLength );
+		super.addResponseHeader( "Content-Range", responseRange.toString() );
+		// TODO: HTTP date format
+		//  Example: Date: Tue, 15 Nov 1994 08:12:31 GMT
+		super.addResponseHeader( "Date", this.getHTTPHandler().getHTTPDateFormat().format(currentDate) );
+
+		// Generate the ETag from inode, resource size and last modification date
+		try {
+		    ETag etag   = ETag.create( resource, uri );
+		    super.addResponseHeader( "ETag", etag.createHeaderValue() ); 
+		} catch( java.security.NoSuchAlgorithmException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 407, 414.